### PR TITLE
Adding amp workspace datasource

### LINKED ISSUE #24503: aws_prometheus_workspace data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It would be very helpful to have a data source for `aws_prometheus_workspace`, particularly so we can use the resulting endpoint URL in other configurations (ex: Prometheus remote writes, Grafana data sources, etc.).

### New or Affected Resource(s)

* aws_prometheus_workspace

### Potential Terraform Configuration

```hcl
data "aws_prometheus_workspace" "test" {
  name = "test-workspace"
} 

locals {
  remote_write_url = "${data.aws_prometheus_workspace.test.endpoint}/api/v1/remote_write"
} 
```

### PR DESCRIPTION
**Pull request by lewinkedrs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #24503.

Output from Datasource (values sanitized):

```
Outputs:

fetched_info_from_aws = {
  "alias" = "west-workspace"
  "arn" = "arn:aws:amp:us-west-2:875375:/workspaces/ws-334a1-2c67-4a1a-94381-9b83d3d73ef7"
  "created_date" = "2022-06-10T18:10:39Z"
  "id" = "ws-31dd8a1-2c67-4a1a-9581-9b83d3d73ef7"
  "prometheus_endpoint" = "https://aps-workspaces.us-west-2.amazonaws.com/workspaces/ws-3da1-2c67-4a1a-9581-9654d3d73ef7/"
  "status" = "ACTIVE"
  "tags" = tomap({
    "terra" = "init"
  })
  "workspace_id" = "ws-31468a1-2c67-4121a-9581-9b83d3d73ef7"
}
```

Output from acceptance testing:

```
✗ make testacc PKG=amp
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 20 -timeout 180m
=== RUN TestAccAMPAlertManagerDefinition_basic
=== PAUSE TestAccAMPAlertManagerDefinition_basic
=== RUN TestAccAMPAlertManagerDefinition_disappears
=== PAUSE TestAccAMPAlertManagerDefinition_disappears
=== RUN TestAccAMPRuleGroupNamespace_basic
=== PAUSE TestAccAMPRuleGroupNamespace_basic
=== RUN TestAccAMPRuleGroupNamespace_disappears
=== PAUSE TestAccAMPRuleGroupNamespace_disappears
=== RUN TestAccAMPWorkspace_basic
=== PAUSE TestAccAMPWorkspace_basic
=== RUN TestAccAMPWorkspace_disappears
=== PAUSE TestAccAMPWorkspace_disappears
=== RUN TestAccAMPWorkspace_tags
=== PAUSE TestAccAMPWorkspace_tags
=== CONT TestAccAMPAlertManagerDefinition_basic
=== CONT TestAccAMPWorkspace_basic
=== CONT TestAccAMPWorkspace_disappears
=== CONT TestAccAMPWorkspace_tags
=== CONT TestAccAMPRuleGroupNamespace_disappears
=== CONT TestAccAMPAlertManagerDefinition_disappears
=== CONT TestAccAMPRuleGroupNamespace_basic
--- PASS: TestAccAMPWorkspace_disappears (19.60s)
```

### CODE REVIEW

File: internal/service/amp/workspace_data_source.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	workspaceID := d.Get("workspace_id").(string)
+	workspace, err := FindWorkspaceByID(conn, workspaceID)
+
+	if err != nil {
+		return fmt.Errorf("error reading AMP Workspace (%s): %w", workspaceID, err)
+	}
+
+	d.SetId(workspaceID)
+
+	workspaceARN := arn.ARN{
```
Comment (jeromeinsf) [inline internal/service/amp/workspace_data_source.go:61]: Is the arn building by hand necessary? https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-APIReference.html#AMP-APIReference-DescribeWorkspace returns the arn generated by the service code

**Comment by jeromeinsf** [comment]
@nicolai86 thoughts?

File: internal/service/amp/workspace_data_source.go (hunk 1 above)
Comment (lewinkedrs) [reply internal/service/amp/workspace_data_source.go:61]: I was curious as well but every data source I've looked at has a similar convention. Here is an example from ebs_volume_data_source.go

```
	arn := arn.ARN{
		Partition: meta.(*conns.AWSClient).Partition,
		Service:   ec2.ServiceName,
		Region:    meta.(*conns.AWSClient).Region,
		AccountID: meta.(*conns.AWSClient).AccountID,
		Resource:  fmt.Sprintf("volume/%s", d.Id()),
	}
	d.Set("arn", arn.String())
```
Seems Arns are treated differently than other describe fields.

Comment (nicolai86) [reply internal/service/amp/workspace_data_source.go:61]: to my best knowledge, the construction of ARNs by hand is only required for services that do not return an ARN in their API response. 
Turns out there are a lot of AWS services that don't return an ARN, and the ARNs are required ie for the AWS resource tagging api.

\cc @jeromeinsf

Comment (lewinkedrs) [reply internal/service/amp/workspace_data_source.go]: Just made a commit to leverage describe workspace for arn just like the other fields.

```
	d.Set("arn", workspace.Arn)
	d.Set("prometheus_endpoint", workspace.PrometheusEndpoint)
```

File: internal/provider/provider.go [hunk 2]
```diff
@@ -416,6 +416,8 @@ func New(_ context.Context) (*schema.Provider, error) {
 			"aws_acmpca_certificate_authority": acmpca.DataSourceCertificateAuthority(),
 			"aws_acmpca_certificate":           acmpca.DataSourceCertificate(),
 
+			"aws_amp_workspace": amp.DataSourceWorkspace(),
```
Comment (jeromeinsf) [inline internal/provider/provider.go:419]: should we stick to the prometheus nomenclature used in https://github.com/lewinkedrs/terraform-provider-aws/blob/main/internal/provider/provider.go#L932 ?

Comment (lewinkedrs) [reply internal/provider/provider.go]: Agreed.

updated to

```
"aws_prometheus_workspace": amp.DataSourceWorkspace(),
```

File: .changelog/26120.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:new-data-source
+aws_amp_workspace
```
Comment (nicolai86) [inline .changelog/26120.txt:2]: this needs to match the name from `provider.go` -> `aws_prometheus_workspace`

File: website/docs/d/amp_workspace.html.markdown [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+subcategory: "AMP (Managed Prometheus)"
+layout: "aws"
+page_title: "AWS: aws_amp_workspace"
```
Comment (nicolai86) [inline website/docs/d/amp_workspace.html.markdown:4]: also rename to match datasource name. recurs.

**Comment by nicolai86** [comment]
let's make sure to carry the right provider name forward in the data source file name as well. `d/amp_workspace.html.markdown` -> `d/prometheus_workspace.html.markdown`. This way it matches the resource file name as well.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccAMPWorkspaceDataSource_' PKG=amp ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/amp/... -v -count 1 -parallel 1  -run=TestAccAMPWorkspaceDataSource_ -timeout 180m
=== RUN   TestAccAMPWorkspaceDataSource_basic
--- PASS: TestAccAMPWorkspaceDataSource_basic (19.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/amp	24.085s
```

**Comment by ewbankkit** [comment]
@lewinkedrs Thanks for the contribution :tada: :clap:.

**Pull request merged**